packages/settings-library/src/settings_library/email.py

```diff
@@ -38,11 +41,19 @@ def both_credentials_must_be_set(cls, values):
     @classmethod
     def enabled_tls_required_authentication(cls, values):
         tls_enabled = values.get("SMTP_TLS_ENABLED")
+        starttls_enabled = values.get("SMTP_STARTTLS_ENABLED")
         username = values.get("SMTP_USERNAME")
         password = values.get("SMTP_PASSWORD")
 
-        if tls_enabled and not (username or password):
+        if (tls_enabled or starttls_enabled) and not (username or password):
             raise ValueError(
                 "when using SMTP_TLS_ENABLED is True username and password are required"
             )
         return values
+
+    @root_validator(pre=True)
+    @classmethod
+    def not_both_tls_and_starttls(cls, values):
+        if values.get("SMTP_TLS_ENABLED") and values.get("SMTP_STARTTLS_ENABLED"):
```

Comment:
when you have multiple exclusive choices it is better if you can a single field (i.e. env variable) associated with an enum that has all the choice. Can you afford that?